hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java

```diff
@@ -911,4 +925,103 @@ public boolean isMaster(Address address) {
+                tlsUsed = addAdvNetworkTlsInfo(tlsSb, e.getKey(), e.getValue().getSSLConfig()) && tlsUsed;
+            }
+        } else {
+            SSLConfig sslConfig = config.getNetworkConfig().getSSLConfig();
+            tlsUsed = addSecurityFeatureCheck(tlsSb, "Use TLS communication protection (Enterprise)",
+                    sslConfig != null && sslConfig.isEnabled());
+        }
+        boolean jetEnabled = config.getJetConfig().isEnabled();
+        if (jetEnabled) {
+            boolean trustedEnv = tlsUsed || !bindAny;
+            addSecurityFeatureCheck(sb, "Use Jet in trusted environments only (single network interface and/or TLS enabled)",
+                    trustedEnv);
+            if (config.getJetConfig().isResourceUploadEnabled()) {
+                addSecurityFeatureInfo(sb, "Jet resource upload is enabled. Any uploaded code can be executed within "
+                        + "the Hazelcast. Use this in trusted environments only.");
+            }
+        } else {
+            addSecurityFeatureInfo(sb, "Jet is disabled");
+        }
+        if (config.getUserCodeDeploymentConfig().isEnabled()) {
+            addSecurityFeatureInfo(sb, "User code deployment is enabled. Any uploaded code can be executed within "
+                    + "the Hazelcast. Use this in trusted environments only.");
+        }
+        addSecurityFeatureCheck(sb, "Disable scripting in the Management Center",
+                !config.getManagementCenterConfig().isScriptingEnabled());
+        SecurityConfig securityConfig = config.getSecurityConfig();
+        boolean securityEnabled = securityConfig != null && securityConfig.isEnabled();
+
+        addSecurityFeatureCheck(sb, "Enable Security (Enterprise)", securityEnabled);
+        if (securityEnabled) {
+            checkAuthnConfigured(sb, securityConfig, "member-authentication", securityConfig.getMemberRealm());
+            checkAuthnConfigured(sb, securityConfig, "client-authentication", securityConfig.getClientRealm());
+        }
+        // TLS here
+        sb.append(tlsSb.toString());
+        PersistenceConfig persistenceConfig = config.getPersistenceConfig();
+        if (persistenceConfig != null && persistenceConfig.isEnabled()) {
+            EncryptionAtRestConfig encryptionAtRestConfig = persistenceConfig.getEncryptionAtRestConfig();
+            addSecurityFeatureCheck(sb, "Enable encryption-at-rest in the Persistence config (Enterprise)",
+                    encryptionAtRestConfig != null && encryptionAtRestConfig.isEnabled());
+        }
+        AuditlogConfig auditlogConfig = config.getAuditlogConfig();
+        addSecurityFeatureCheck(sb, "Enable auditlog (Enterprise)", auditlogConfig != null && auditlogConfig.isEnabled());
+
+        sb.append("\nCheck the hazelcast-security-hardened.xml/yaml example config file to find why and how to configure"
```

Comment:
When you use the `hazelcast-security-hardened.xml` do all the checks actually light up in green?

Reply:
The hardened files are intended for OS use too. It means enterprise features are not enabled. The sections are still included in the hardened-security config files with comments, so enabling EE features should be easy for users.

When running:
```bash
export JAVA_OPTS=-Dcom.hazelcast.system.security
bin/hz-start
```

then the console shows

![screenshot](https://user-images.githubusercontent.com/962661/133451521-e2241e8e-8cd8-4307-a6e3-9f73bb521a38.png)

Reply:
Very nice :+1: